crates/extensions/tedge_log_manager/src/config.rs

```diff
@@ -87,3 +89,317 @@ impl LogManagerConfig {
         })
     }
 }
+
+/// Plugin filtering configuration parsed from tedge-log-plugin.toml
+#[derive(Clone, Deserialize, Debug, Default, PartialEq, Eq)]
+pub struct PluginConfig {
+    #[serde(default)]
+    pub plugins: HashMap<String, Vec<PluginFilterEntry>>,
+}
+
+/// Individual filter entry for a plugin
+#[derive(Clone, Deserialize, Debug, PartialEq, Eq)]
+pub struct PluginFilterEntry {
+    #[serde(default)]
+    pub include: Option<String>,
+    #[serde(default)]
+    pub exclude: Option<String>,
+}
```

Comment:
Looking how this is used, an `enum` seems more appropriate.

```suggestion
pub enum PluginFilterEntry {
     Include(String),
     Exclude(String),
}
```

I would even go a step further with:


```suggestion
pub enum PluginFilterEntry {
     Include(glob::Pattern),
     Exclude(glob::Pattern),
}
```

Reply:
After [this](https://github.com/thin-edge/thin-edge.io/pull/3838#discussion_r2492813337) comment from  @reubenmiller I'm having second thoughts on converting that struct definition into an enum as it prevents users from defining both `include` and `exclude` filters under the same plugin entry as follows:

 ```
 [[plugins.journald]]
exclude = "systemd-*"
include = "systemd-logind"
```

The struct might be more future-proof as well if we want to support more fields beyond `include` and `exclude` in future.

Reply:
My suggestion is not to prevent a combination of include and exclude rules for a given plugin, but to simplify the representation and the implementation logic. Note that currently a `Vec<PluginFilterEntry>>` is attached to each log type, making the enum enough and clearer. Two different structs might be required, though. A first one for the TOML file, aka `PluginConfig`, a second one to control the extent of log types to manage.

Reply:
Have done the compilation to regex pattern in [2d9f927](https://github.com/thin-edge/thin-edge.io/pull/3838/commits/2d9f9274e2ce50737d30f15de11a32fa0a05e618). The enum couldn't be done due to the possibility of both `include` and `exclude` filters under the `[[plugins.journald]]` entry, unless we use an enum like:
```
enum Filter {
   Include(Regex)
   Exclude(Regex)
   Both(Regex)
}
```
which would make little difference in the filtering logic.

Reply:
Okay let's keep the struct.

However, I disagree with the rationale. The in-memory representation of the filters doesn't have to match the external schema of the configuration file. We could have a vector of enum values with two cases or two vectors (one for the included regex the other for the excluded ones)

Reply:
Resolved that as well in 9d4bbb63d1 (squashed commit) with a `TomlPluginConfig` struct used for parsing and a compiled `PluginConfig` struct that maintains `include_patterns` and `exclude_patterns` separately.